Source/WebKit/Shared/API/glib/WebKitContextMenuItem.cpp

```diff
@@ -357,6 +360,32 @@ WebKitContextMenuAction webkit_context_menu_item_get_stock_action(WebKitContextM
 #endif // ENABLE(CONTEXT_MENUS)
 }
 
+#if PLATFORM(WPE)
+/**
+ * webkit_context_menu_item_get_title:
+ * @item: a #WebKitContextMenuItem
+ *
+ * Gets the title of @item.
+ *
+ * Returns: the title of @item, or %NULL if @item is a separator.
+ *
+ */
+const gchar* webkit_context_menu_item_get_title(WebKitContextMenuItem* item)
+{
+    g_return_val_if_fail(WEBKIT_IS_CONTEXT_MENU_ITEM(item), nullptr);
+
+#if ENABLE(CONTEXT_MENUS)
+    if (item->priv->menuItem->type() == WebCore::ContextMenuItemType::Separator)
+        return nullptr;
+    if (item->priv->titleUTF8.isNull())
+        item->priv->titleUTF8 = item->priv->menuItem->title().utf8();
+    return item->priv->titleUTF8.data();
+#else
+    g_assert_not_reached();
+#endif // ENABLE(CONTEXT_MENUS)
```

Comment:
I think we could remove the preprocessor guards, and always build with `ENABLE_CONTEXT_MENUS`. At the moment only the PlayStation port has this build option disabled, so I think we could remove the option, and change the guards needed for PlayStation builds to `PLATFORM(PLAYSTATION)` instead. The cleanup to remove the build option can be done in a follow-up patch, for this patch I think we want to assume that the option will always be enabled at build time.


```suggestion
    if (item->priv->menuItem->type() == WebCore::ContextMenuItemType::Separator)
        return nullptr;
    if (item->priv->titleUTF8.isNull())
        item->priv->titleUTF8 = item->priv->menuItem->title().utf8();
    return item->priv->titleUTF8.data();
```

Reply:
We have those guards everywhere...